refresh-i18n/src/Main.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ViewPatterns        #-}
+
+-- | usage: "$0 [target_path]"
+--
+-- (I tried to use packages purescript, language-javascript, and language-purescript, but it was a
+-- lot easier to do just stupid pattern matching on the source files without any parsing.)
+module Main
+where
+
+import Control.Exception (assert)
+import Control.Monad (void)
+import Data.Function (on)
+import Data.Functor.Infix ((<$$>))
+import Data.List (foldl', union, sort, nub, groupBy)
+import Data.Maybe (listToMaybe, catMaybes)
+import Data.Monoid ((<>))
+import Data.String.Conversions (ST, cs)
+import System.Directory (doesDirectoryExist, setCurrentDirectory, getDirectoryContents)
+import System.Environment (lookupEnv, getArgs)
+import System.FilePath ((</>), (<.>), takeExtension)
+import System.IO (hPutStrLn, stderr)
+import System.Process (system)
+
+import qualified Data.Text as ST
+import qualified Data.Text.IO as ST
+
+
+i18nModule :: FilePath
+i18nModule = "I18n"
+
+main :: IO ()
+main = do
+    setCurrentDirectoryToTarget
+    purss :: [FilePath] <- getTranslateableFiles "."
+    transKeys :: [ST] <- nub . sort . foldl' union [] <$> mapM translationKeys purss
```

Comment:
Unless I'm missing something, this is *very* inefficient. `union` has at least quadratic complexity, so the whole thing is O(n^3) or worse. If this becomes a problem, it should be easy to fix by using `Data.Set`